lms/views/decorators/h_api.py

```diff
@@ -61,8 +63,14 @@ def wrapper(request, jwt, context=None):
             ],
         }
 
-        # Call the h API to create the user in h if it doesn't exist already.
-        request.find_service(name="hapi").post("users", user_data, statuses=[409])
+        # Send a PATCH user request to update user data to keep it in sync
+        # with the data sent over LTI. This will succeed if the user already
+        # exists.
+        try:
+            hapi_service.patch(f"users/{context.h_username}", {"display_name": context.h_display_name})
+        except HAPINotFoundError:
+            # Call the h API to create the user in h if it doesn't exist already.
+            hapi_service.post("users", user_data, statuses=[409])
```

Comment:
I have a hunch we may no longer want to squash 409s here, as this request should only get fired if the user doesn't exist. Thoughts on how we should handle 409s?

Reply:
A 409 is still possible if two concurrent requests from the lms app are racing.

Reply:
I guess a `409` could happen for a few reasons:

1. This user (same username, authority and identities) has already been created. In this case we should be able to ignore the `409` and move on. I think a user upsert endpoint would fix this issue (and group upsert is currently fixing it for groups). I think this will be pretty rare because it requires two concurrent requests from the same user.

2. Something has gone wrong in the DB. For example a user with this username _doesn't_ exist yet, but there _is_ a user with a matching identity. Really bad! AFAIK there's no way that this can happen but a bug (present or future) in the LMS app or h could cause it.

3. A bug (present or future) in h causes h to incorrectly send a `409`

4. In the future h changes so that there are new ways for users to conflict and the LMS app falls foul of these changes

5. An email conflict -- two users in the same authority with different usernames and identities but the same email. This can't happen currently because the LMS app does not add emails to users. But I could imagine future changes triggering this.

6. Incorrect dev env

7. Something else...?

Based on the above I'd like to shy away from silencing `409`s here. Lets allow a `409` to crash the LMS app, which will show the user an error page and report the issue to Sentry. If the reason for the `409` was 1 above the user only needs to try again and it should work, but I think this will be very rare. And this way if a `409` happens for any other reason now or in the future it'll crash right away, rather than continuing and potentially doing the wrong thing or crashing later, and we'll hear about it in Sentry

Reply:
```suggestion
            hapi_service.post("users", user_data)
```